Выполнить задание 7го спринта.

fcevt opened the pull request.

fcevt committed:
Выполнить задание 7го спринта.

pattyku commented on src/com/yandex/tracker/servise/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,208 @@
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
в сервисах не переопределяем equals и hashCode

pattyku commented on src/com/yandex/tracker/servise/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,208 @@
+        tasksFile = new File(System.getProperty("user.dir"),"tasks.txt");
+    }
+
+    public FileBackedTaskManager(int id, HashMap<Integer, Task> tasks,HashMap<Integer, Epic> epics,
```
наследуется от InMemoryTaskManager, нужно использовать его поля
```
public FileBackedTaskManager(File file) {
    this. tasksFile = file;
}
```

pattyku commented on src/com/yandex/tracker/servise/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,208 @@
+
+    //создание менеджера из файла
+    public static FileBackedTaskManager loadFromFile(File file) throws ManagerReadException {
+        HashMap<Integer, Task> tasks = new HashMap<>();
```
мапы удаляем
`final FileBackedTaskManager taskManager = new FileBackedTaskManager(file);`
и дальше используем taskManager для получения задач/подзадач/эпиков

pattyku commented on src/com/yandex/tracker/servise/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,208 @@
+
+    //создание тасок из файла в зависимости от типа
+    private static Task taskFromArray(String[] array) {
+        if (array[1].equals(TaskType.TASK.name())) {
```
В работе часто читаешь код и если вернутся к этому коду через несколько месяцев, не сразу поймешь что стоит за array[1], array[2] и т.д. 
Лучше вынести их в переменные
```
final int id = Integer.parseInt(array[0]);
final TaskType type = TaskType.valueOf(array[1]);
final String name = array[2];
```

pattyku commented on src/com/yandex/tracker/servise/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,208 @@
+package com.yandex.tracker.servise;
```
Также в классе  TaskManager в методе getDefault() нужно поменять реализацию на FileBackedTaskManager
```
public static TaskManager getDefault() {
        return new FileBackedTaskManager(new File(...));
}
```

fcevt committed:
Удалить equals и hashCode из FileBackedTaskManager.
Переписать тесты в связи с удалением equals и hashCode из FileBackedTaskManager.
Удалить новые конструкторы из менеджеров.
Поменять способ восстановления менеджера из файла с использования конструктора на использование геттеров и сеттеров.
Создать в методе taskFromArray() понятные переменные для создания тасок.
 Поменять в классе Managers реализацию метода getDefault() чтобы он возвращал FileBackedTaskManager

pattyku commented on src/com/yandex/tracker/servise/Managers.java:
```diff
@@ -1,12 +1,14 @@
 
-    public static TaskManager getDefault() {
-        return new InMemoryTaskManager();
+    public static FileBackedTaskManager getDefault() {
```
возвращаемый тип оставляем  TaskManager
public static TaskManager getDefault() {
            return new FileBackedTaskManager(new File(System.getProperty("user.dir"), "tasksFile.txt"));
}

fcevt committed:
Исправить в классе Managers реализацию метода getDefault() чтобы возвращаемый тип был TaskManager.
